main.cpp

```diff
@@ -28,57 +28,47 @@ struct T
 
 struct Struct1                                //4
 {
-    T* compare(T* a, T* b) //5
+    const T& compare(const T& a, const T& b) //5
```

Comment:
Follow the instructions
```
        hint: There is no reference equivalent to nullptr.  
        if a pointer (including nullptr) is being returned anywhere, don't try to convert it to a reference.
```
- [ ] revise

Reply:
I changed to
`T* compare( T& a, T& b) `

If I'm code this
`T* compare(const T& a,const T& b) `
I get this warning
`main.cpp:33:40: error: cannot initialize return object of type 'T *' with an
      rvalue of type 'const T *'
        if( a.value < b.value ) return &a;`

I didn't understand the reason...